src/book/components/BookOverview/BookOverview.test.tsx

```diff
@@ -48,55 +57,41 @@ describe("Book Overview Component", () => {
     expect(authorsColumn).toBeInTheDocument();
     expect(titleColumn).toBeInTheDocument();
   });
-  it("renders the master table rows", async () => {
+  it("Renders books table with data received from server", async () => {
     // given
     expect.hasAssertions();
-    act(() => {
-      render(<BookOverview />, { wrapper });
-    });
-
-    // when
-    return bookServiceMockPromise.then(() => {
-      const johnExamleRow = screen.getByText(/John Example/i);
-      const joeSmithRow = screen.getByText(/Joe Smith/i);
-      // then
-      expect(johnExamleRow).toBeInTheDocument();
-      expect(joeSmithRow).toBeInTheDocument();
-    });
+    
+    render(<BookOverview />, { wrapper: WrapperComponent });
+    expect(await screen.findByText(/Julius Verne/i)).toBeInTheDocument();
+    expect(await screen.findByText(/Frank Herbert/i)).toBeInTheDocument();
+      
   });
-  it("renders details upon click on the row", () => {
+  it("renders details upon click on the row", async () => {
     // given
-    act(() => {
-      render(<BookOverview />, { wrapper });
-      jest.runAllTimers();
-    });
+    render(<BookOverview />, { wrapper: WrapperComponent });
     // when
-    return bookServiceMockPromise.then(() => {
-      const row = screen.getByText(/John Example/i).closest("tr");
-      row && fireEvent.click(row);
-      // then
-      expect(screen.getByText(/Authors:/i)).toBeInTheDocument();
-    });
+    const row = (await screen.findByText(/Julius Verne/i)).closest("tr");
+    row && userEvent.click(row);
+    // then
+    expect(screen.getByText(/Authors:/i)).toBeInTheDocument();
   });
 
-  it("updates a book row upon changes done in the details", () => {
+  it("updates a book row upon changes done in the details", async () => {
     // given
-    act(() => {
-      render(<BookOverview />, { wrapper });
-      jest.runAllTimers();
-    });
+    render(<BookOverview />, { wrapper: WrapperComponent });
     // when
-    return bookServiceMockPromise.then(() => {
-      const row = screen.getByText(/John Example/i).closest("tr");
-      row && fireEvent.click(row);
-      const newAuthor = "New Author";
-      const authors = screen.getByLabelText(/Authors:/i);
-      fireEvent.change(authors, { target: { value: newAuthor } });
-      const form = authors.closest("form");
-      form && fireEvent.submit(form, { preventDefault: jest.fn() });
-      row?.querySelector("td");
-      const updatedAuthorCell = row?.querySelector("td");
-      expect(updatedAuthorCell).toHaveTextContent(newAuthor);
-    });
+
+    const row = (await screen.findByText(/Julius Verne/i)).closest("tr");
+    row && userEvent.click(row);
+    const newAuthor = "New Author";
+    const authors = screen.getByLabelText(/Authors:/i);
+    userEvent.clear(authors);
```

Comment:
i decided to use userEvent events instead of fireEvent, as they resemble more native events in browser and are also easier to teach :)